source/standard_scores.Rmd

```diff
@@ -0,0 +1,1515 @@
+
+Here we start using the Pandas library to load table data into Python.
+
+Thus far we have used the Numpy library to work with data in arrays.  Pandas is
+As always with Python, when we want to use a library, we have to `import` it
+first.
+
+We have used the term *library* here, but Python uses the term *module* to
+refer to libraries of code and data that you `import`.
+
+When using Numpy, we write:
+
+```{python}
+# Import the Numpy library (module), name it "np".
+import numpy as np
+```
+
+Now we will use the Pandas library (module).
+
+We can import Pandas like this:
+
+```{python}
+# Import the Pandas library (module)
+import pandas
+```
+
+As Numpy has a standard abbreviation `np`, that almost everyone writing Python
+code will recognize and use, so Pandas has the standard abbreviation `pd`:
+
+```{python}
+# Import the Pandas library (module), name it "pd".
+import pandas as pd
+```
+
+Pandas is the standard data science library for Python.   It is particularly
+good at loading data files, and presenting them to us as a useful table-like
+structure, called a *data frame*.
+
+We start by using Pandas to load our data file:
+
+```{python}
+district_income = pd.read_csv('data/congress_2023.csv')
+```
+
+We have thus far done many operations that returned Numpy *arrays*.
+`pd.read_csv` returns a Pandas *data frame*:
+
+```{python}
+type(district_income)
+```
+
+A data frame is Pandas' own way of representing a table, with columns and rows.
+You can think of it as Python's version of a spreadsheet.  As strings or Numpy
+arrays have *methods* (functions attached to the array values), so Pandas data
+frames have methods.  These methods do things with the data frame to which they
+are attached.  For example, the `head` method of the data frame shows (by
+default) the first five rows in the table:
+
+```{python}
+# Show the first five rows in the data frame
+district_income.head()
+```
+
+The data are in income order, sorted lowest to highest, so the first five
+districts are those with the lowest household income.
+
+We are particularly interested in the column named `Median_Income`.
+
+You may remember the idea of *indexing*, introduced in @sec-array-indexing.
+Indexing occurs when we fetch data from within a container, such as a string or
+an array.  We do this by putting square brackets `[]` after the value we want
+to index into, and put something inside the brackets to say what we want.
+
+For example, to get the *first* element of the `priv` array above, we use
+indexing:
+
+```{python}
+# Fetch the first element of the priv array with indexing.
+# This is the element at position 0.
+priv[0]
+```
+
+As you can index into strings and Numpy arrays, by using square brackets, so
+you can index into Pandas data frames.  Instead of putting the *position*
+between the square brackets, we can put the *column name*.  This fetches the
+data from that column, returning a new thing called a Pandas *Series*.
```

Comment:
Do we want to introduce the concept of an `object` at some point (instead of `thing`!)?